backend/inspirehep/disambiguation/cli.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import click
+from elasticsearch_dsl import Q
+from inspire_utils.record import get_values_for_schema
+from invenio_db import db
+
+from inspirehep.records.api import AuthorsRecord
+from inspirehep.records.models import RecordsAuthors
+from inspirehep.search.api import AuthorsSearch
+
+
+@click.group()
+def disambiguation():
+    """Commands for disambiguation"""
+
+
+@disambiguation.command()
+def clean_stub_authors():
+    """Removes all the authors created by disambiguation and having no linked papers."""
+    # We get all the stub authors (created by disambiguation) from ES and we verify
+    # in db if the returned records are stub (ES data might be outdated)
+    stub_authors_query = Q("term", stub=True)
+    stub_authors = (
+        AuthorsSearch().query(stub_authors_query).source(["control_number"]).execute()
+    )
+    stub_authors_control_numbers = [
+        ("aut", str(author["control_number"])) for author in stub_authors
+    ]
+    # We change isolation level in db to the higher one (serializable) to avoid
+    # issues with race condition
+    db.session.connection(execution_options={"isolation_level": "SERIALIZABLE"})
+    stub_authors_bais = {
+        get_values_for_schema(author["ids"], "INSPIRE BAI")[0]
+        for author in AuthorsRecord.get_records_by_pids(stub_authors_control_numbers)
+    }
+    # We verify which authors have linked papers
+    stub_authors_with_papers = set(
+        query_authors_with_linked_papers_by_bai(stub_authors_bais)
+    )
+    # For every author who has not linked papers we delete record
+    for author_bai in stub_authors_bais.difference(stub_authors_with_papers):
+        author = AuthorsRecord.get_record_by_pid_value(author_bai, "bai")
+        author.delete()
```

Comment:
Does this remove the BAI from the PIDstore or mark it as deleted? I guess we want to actually remove the BAI from the PIDstore, otherwise we'll get ugly things like `Z.Wang.4235` at some point.

Reply:
Doesn't it remove bai too?
https://github.com/inspirehep/inspirehep/blob/cb15e67376042277a0550a2ae0f8b7147a8d65ae/backend/inspirehep/records/api/base.py#L501

Reply:
Looks like it, I hadn't checked :+1: https://github.com/inspirehep/inspirehep/blob/cb15e67376042277a0550a2ae0f8b7147a8d65ae/backend/inspirehep/pidstore/providers/bai.py#L117-L119